records/generator.go

```diff
@@ -33,10 +33,15 @@ type RecordGenerator struct {
 
 // ParseState retrieves and parses the Mesos master /state.json and converts it
 // into DNS records.
-func (rg *RecordGenerator) ParseState(leader string, c Config) error {
+func (rg *RecordGenerator) ParseState(masters []string, c Config) error {
+	var leader string
+
+	if len(masters) > 0 {
+		leader = masters[0]
+	}
 
 	// find master -- return if error
-	sj, err := rg.findMaster(leader, c.Masters)
```

Comment:
What I meant was that `findMaster` would become: `func (rg RecordGenerator) findMaster(masters ...string) (state.State, error)`